### PR TITLE
Add CUSTOM_COMPILE_COMMAND support to uv pip compile

### PR DESCRIPTION
**Pull request by BakerNet**
## Summary

This adds support for `CUSTOM_COMPILE_COMMAND` support to change the header comment in generated requirements files.

See Issue:  
- #1527 

From [pip-tools docs](https://pip-tools.readthedocs.io/en/latest/):

> You might be wrapping the pip-compile command in another script. To avoid confusing consumers of your custom script you can override the update command generated at the top of requirements files by setting the CUSTOM_COMPILE_COMMAND environment variable.

## Test Plan

See unit test included

### CODE REVIEW

File: crates/uv/src/commands/pip_compile.rs [hunk 1]
```diff
@@ -369,7 +370,15 @@ pub(crate) async fn pip_compile(
...
             writer,
             "{}",
-            format!("#    {}", cmd(include_index_url, include_find_links,)).green()
+            format!(
+                "#    {}",
+                cmd(
+                    include_index_url,
+                    include_find_links,
+                    custom_compile_command
+                )
```
Comment (BakerNet) [inline crates/uv/src/commands/pip_compile.rs:379]: Alternatively, if you prefer to have it inline:
```suggestion
                custom_compile_command.unwrap_or(cmd(include_index_url, include_find_links))
```
and remove the changes in `cmd` function

File: crates/uv/src/commands/pip_compile.rs [hunk 2]
```diff
@@ -435,7 +444,14 @@ pub(crate) async fn pip_compile(
...
 #[allow(clippy::fn_params_excessive_bools)]
-fn cmd(include_index_url: bool, include_find_links: bool) -> String {
+fn cmd(
+    include_index_url: bool,
+    include_find_links: bool,
+    custom_compile_command: Option<String>,
+) -> String {
+    if let Some(cmd_str) = custom_compile_command {
+        return cmd_str;
+    }
```
Comment (BakerNet) [inline crates/uv/src/commands/pip_compile.rs:454]: If above suggestion, revert
```suggestion
fn cmd(include_index_url: bool, include_find_links: bool) -> String {
```

File: crates/uv/src/main.rs [hunk 3]
```diff
@@ -324,6 +324,10 @@ struct PipCompileArgs {
     #[clap(long)]
     no_header: bool,
 
+    /// Change header comment to reflect custom command wrapping `uv pip compile`.
+    #[clap(long, env = "CUSTOM_COMPILE_COMMAND")]
```
Comment (BakerNet) [inline crates/uv/src/main.rs:328]: Alternate env name:
```suggestion
    #[clap(long, env = "UV_CUSTOM_COMPILE_COMMAND")]
```

File: crates/uv/tests/pip_compile.rs [hunk 4]
```diff
@@ -3482,6 +3482,68 @@ fn no_header() -> Result<()> {
...
+
+    ----- stderr -----
+    Resolved 6 packages in [TIME]
+    "###
+    );
+
+    // with env var
+    uv_snapshot!(context.compile()
+            .arg("requirements.in")
+            .env("CUSTOM_COMPILE_COMMAND", "./custom-uv-compile.sh"), @r###"
```
Comment (BakerNet) [inline crates/uv/tests/pip_compile.rs:3521]: Alternat env name:
```suggestion
            .env("UV_CUSTOM_COMPILE_COMMAND", "./custom-uv-compile.sh"), @r###"
```

**Review by konstin — APPROVED** [review]
Thank you!

**Comment by henryiii** [comment]
This adds `UV_CUSTOM_COMPILE_COMMAND`, not `CUSTOM_COMPILE_COMMAND`. Both the PR and the release notes state the `CUSTOM_COMPILE_COMMAND` one, which is the same as pip compile, but this was added with a `UV_` prefix.

**Comment by henryiii** [comment]
(I read the release notes, then couldn't figure out why `CUSTOM_COMPILE_COMMAND` that was previously set for `pip-compile` wasn't now working for uv until I read the diff)

**Comment by konstin** [comment]
Thanks, i updated the release notes.

**Pull request merged**